sunpy/database/database.py

```diff
@@ -121,6 +121,23 @@ def __str__(self):
         return errmsg.format(self.database_entry, self.tag_name)
 
 
+class PartialFetchError(Exception):
+    """This exception is raised if the number of files returned by the
+    downloader does not match the number expected.
+
+    """
+
+    def __init__(self, paths, expected):
```

Comment:
`n_expected` might be a clearer variable name - I originally thought expected might be a list of expected paths or similar.